src/Utility/TensorFunction.h

```diff
@@ -49,6 +50,23 @@ class TensorFunc : public utl::SpatialFunction<Tensor>, public FunctionBase
   {
     return this->evaluate(X).trace();
   }
+
+  //! \brief Evaluates first derivatives of the function.
+  utl::matrix3d<Real> gradient(const Vec3& X) const
+  {
+    const size_t nsd = sqrt(this->dim());
```

Comment:
`dim()` just returns the protected member `ncmp` so you could just write `sqrt(ncmp)` here.

Reply:
ok.